File: src/main/angular/src/app/core/services/item.service.ts
import { Injectable } from '@angular/core';
import {Item} from '../dto/Item';
import {Observable} from 'rxjs';
import {HttpClient} from '@angular/common/http';

export const MAIN_URL = 'http://localhost:8080';
const ITEM_URL = '/api/item';

@Injectable({
  providedIn: 'root'
})
export class ItemService {

  constructor(private httpClient: HttpClient) { }

  getAllItems(): Observable<Array<Item>>{
    return this.httpClient.get<Array<Item>>(MAIN_URL + ITEM_URL + '/all');
  }
  
  getCount(): Observable<number>{
    return this.httpClient.get<number>(MAIN_URL + ITEM_URL + '/count');
  }

  saveItem(item: Item): Observable<boolean>{
    return this.httpClient.post<boolean>(MAIN_URL + ITEM_URL + '/save', item);
  }

  updateItem(item: Item): Observable<boolean>{
    return this.httpClient.put<boolean>(MAIN_URL + ITEM_URL + '/update', item);
  }

  deleteItem(item: Item): Observable<boolean>{
    return this.httpClient.delete<boolean>(MAIN_URL + ITEM_URL + '/delete/'+ item.code);
  }
}

File: src/main/java/com/jamith/absd/service/ItemService.java
package com.jamith.absd.service;

import com.jamith.absd.dto.ItemDto;
import com.jamith.absd.entity.Item;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface ItemService {

    List<ItemDto> getAllItems();

    boolean saveItem(ItemDto dto);

    boolean updateItem(ItemDto dto);

    boolean deleteItem(String code);

    Item getItemByCode(String code);

    boolean updateItemQty(Integer qty, String code);

    Integer getCount();
}
